examples/tree_2d_dgsem/elixir_diffusion_steady_state_linear_map.jl

```diff
@@ -0,0 +1,107 @@
+using Trixi
+
+###############################################################################
+
+# Build pure diffusion (Laplace) operator
+advection_velocity = (0, 0)
+equations = LinearScalarAdvectionEquation2D(advection_velocity)
+diffusivity() = 1
+equations_parabolic = LaplaceDiffusion2D(diffusivity(), equations)
+
+# The hyperbolic flux does not matter for this example since
+# the hyperbolic part is zero.
+solver = DGSEM(polydeg = 5, surface_flux = flux_central)
+
+coordinates_min = (0.0, 0.0)
+coordinates_max = (1.0, 1.0)
+mesh = TreeMesh(coordinates_min, coordinates_max,
+                initial_refinement_level = 2,
+                n_cells_max = 80_000,
+                periodicity = false)
+
+# Analytical/continuous steady-state solution
+function continuous_solution(x, t, equations)
+    a = (1 - cosh(2 * pi)) / (sinh(2 * pi))
+
+    u_sol = (cosh(2 * pi * x[2]) + a * sinh(2 * pi * x[2])) * sinpi(2 * x[1])
+    return SVector(u_sol)
+end
+initial_condition = continuous_solution
+
+function bc_homogeneous(x, t, equations)
+    return SVector(0)
+end
+bc_homogeneous_dirichlet = BoundaryConditionDirichlet(bc_homogeneous)
+
+function bc_sin(x, t, equations)
+    return SVector(sinpi(2 * x[1]))
+end
+bc_sin_dirichlet = BoundaryConditionDirichlet(bc_sin)
+
+# Same boundary conditions for hyperbolic and parabolic part
+boundary_conditions = (; x_neg = bc_homogeneous_dirichlet,
+                       y_neg = bc_sin_dirichlet,
+                       y_pos = bc_sin_dirichlet,
+                       x_pos = bc_homogeneous_dirichlet)
+
+# `solver_parabolic = ViscousFormulationLocalDG()` strictly required for elliptic/diffusion-dominated problem
+semi = SemidiscretizationHyperbolicParabolic(mesh,
+                                             (equations, equations_parabolic),
+                                             initial_condition, solver;
+                                             solver_parabolic = ViscousFormulationLocalDG(),
+                                             boundary_conditions = (boundary_conditions,
+                                                                    boundary_conditions))
+
+# Note that `linear_structure` does not access the `initial_condition`/steady-state solution
+A_map, b = linear_structure(semi)
+
+# Direct solve, with explicit sparse matrix construction.
+# Has trouble due to poor conditioning, visible in top right corner
+#=
+using SparseArrays
+A_matrix = sparse(A_map)
+u_ls = A_matrix \ b
+=#
```

Comment:
Is the iteratively obtained numerical solution much better than the one obtained by a sparse direct solve?

Reply:
Yes, the direct solution is trash:

<img width="1084" height="945" alt="Screenshot from 2025-10-20 14-33-22" src="https://github.com/user-attachments/assets/a58080a8-9952-4100-8f25-c1326d9238ca" />

VS the iterative solver

<img width="1084" height="945" alt="Screenshot from 2025-10-20 14-34-24" src="https://github.com/user-attachments/assets/fced17b2-15dd-4e1d-b1c0-3de6423f4e06" />